File: src/main/java/com/sfh/pokeRogueBot/model/browser/gamejson/CurrentBattle.java
package com.sfh.pokeRogueBot.model.browser.gamejson;

import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
public class CurrentBattle {

    private Trainer trainer;
    private Map<String, Object> seenEnemyPartyMemberIds;
    private int turn;
    private Map<String, Object> turnCommands;
    private int enemySwitchCounter;
    private int escapeAttempts;
    private Object lastUsedPokeball;
    private int moneyScattered;
    private Map<String, Object> playerParticipantIds;
    private List<Object> postBattleLoot;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/browser/gamejson/CurrentBattle.kt
package com.sfh.pokeRogueBot.model.browser.gamejson

data class CurrentBattle(
    val trainer: Trainer?,
    val seenEnemyPartyMemberIds: Map<String, Any>?,
    val turn: Int,
    val turnCommands: Map<String, Any>?,
    val enemySwitchCounter: Int,
    val escapeAttempts: Int,
    val lastUsedPokeball: Any?,
    val moneyScattered: Int,
    val playerParticipantIds: Map<String, Any>?,
    val postBattleLoot: List<Any>?
)